packages/itwin/grouping-mapping-widget/src/widget/components/SelectIModel.tsx

```diff
@@ -26,20 +27,29 @@ const SelectIModel = ({
   onCancel,
   backFn,
 }: SelectIModelProps) => {
-  const apiContext = useContext(ApiContext);
+  const { getAccessToken, prefix } = useGroupingMappingApiConfig();
+  const [accessToken, setAccessToken] = useState<AccessToken>();
 
   const apiOverrides = useMemo<ApiOverrides<IModelFull[]>>(
```

Comment:
Just a note about `useMemo`: https://reactjs.org/docs/hooks-reference.html#usememo
`You may rely on useMemo as a performance optimization, not as a semantic guarantee.`

This place looks like you are relying on `useMemo` as a semantic guarantee to prevent re-renderings.

Reply:
Removed useMemo. I vaguely remember people screaming about this.

Reply:
Haha. It wasn't directed to me. There was some discussion on the documentation being a bit contradictory. 
https://reactjs.org/docs/hooks-faq.html#what-can-i-do-if-my-effect-dependencies-change-too-often
`As a last resort, you can add a function to effect dependencies but wrap its definition into the` [useCallback](https://reactjs.org/docs/hooks-reference.html#usecallback)` Hook. This ensures it doesn’t change on every render unless its own dependencies also change:`

In the `useCallback` docs it says `useCallback(fn, deps) is equivalent to useMemo(() => fn, deps).` So inherently its also subject to future cache busting. So it is a little confusing to suggest this pattern.